File: app/src/main/java/com/stm/evenote/rsync/model/OperationFactory.java
package com.stm.evenote.rsync.model;

public interface OperationFactory {

    Operation noOp(SyncFile file);

    Operation updateFile(SyncFile fileToUpdate, SyncFile fileWithNewContent);

    Operation delete(SyncFile fileToDelete);

    Operation newFile(SyncFile targetFile);
}

File: app/src/main/java/com/stm/evenote/rsync/model/OperationFactory.kt
package com.stm.evenote.rsync.model

interface OperationFactory {
    fun noOp(file: SyncFile): Operation
    fun updateFile(fileToUpdate: SyncFile, fileWithNewContent: SyncFile): Operation
    fun delete(fileToDelete: SyncFile): Operation
    fun newFile(newFile: SyncFile): Operation
}